lib/matplotlib/font_manager.py

```diff
@@ -1399,38 +1399,55 @@ def _find_fonts_by_props(self, prop, fontext='ttf', directory=None,
 
         """
 
-        rc_params = tuple(tuple(rcParams[key]) for key in [
+        rc_params = tuple(tuple(mpl.rcParams[key]) for key in [
             "font.serif", "font.sans-serif", "font.cursive", "font.fantasy",
             "font.monospace"])
 
         prop = FontProperties._from_any(prop)
 
         fpaths = []
         for family in prop.get_family():
-            cprop = prop.copy()
-
-            # set current prop's family
-            cprop.set_family(family)
-
-            # do not fall back to default font
-            try:
-                fpaths.append(
-                    self._findfont_cached(
-                        cprop, fontext, directory,
-                        fallback_to_default=False,
-                        rebuild_if_missing=rebuild_if_missing,
-                        rc_params=rc_params,
-                    )
-                )
-            except ValueError:
-                if family in font_family_aliases:
-                    _log.warning(
-                        "findfont: Generic family %r not found because "
-                        "none of the following families were found: %s",
-                        family,
-                        ", ".join(self._expand_aliases(family))
+            if family in font_family_aliases:
+                # deal with the sans/san-serif/sans serif issue
```

Comment:
This and the following line (the start of the for loop) can be just
```python
for fam in self._expand_aliases(family):
    if fam in font_family_aliases: continue  # is that even possible?
    ...
```

Reply:
This will go into another PR, this change seems to be causing a 10x memory increase and I do not understand why yet.